src/utility.jl

```diff
@@ -0,0 +1,80 @@
+
+# TODO: 
+# - tidy up this file
+
+
+Base.zero(lv::LorentzVector{CS,T}) where {CS,T} = LorentzVector{CS}(ntuple(i->zero(T),4)...)
+
+
+Base.:(==)(lv1::LorentzVector, lv2::LorentzVector) = false
+
+function Base.:(==)(lv1::LorentzVector{CS}, lv2::LorentzVector{CS}) where {CS<:LorentzVectorBase.AbstractCoordinateSystem}
+    return lv1.comp1 ==lv2.comp1 && lv1.comp2 ==lv2.comp2 && lv1.comp3 ==lv2.comp3 && lv1.comp0 ==lv2.comp0
+end
+
+Base.isapprox(lv1::LorentzVector,lv2::LorentzVector; args...) = false
+
+function Base.isapprox(
+    lv1::LorentzVector{CS}, 
+    lv2::LorentzVector{CS}; 
+    atol::Real=0, 
+    rtol::Real=atol>0 ? 0 : sqrt(eps())
+) where {CS<:LorentzVectorBase.AbstractCoordinateSystem}
+    return isapprox(lv1.comp1,lv2.comp1;atol,rtol) && isapprox(lv1.comp2,lv2.comp2;atol,rtol) && isapprox(lv1.comp3,lv2.comp3;atol,rtol) && isapprox(lv1.comp0,lv2.comp0;atol,rtol) 
+end
+
+
+# phi is in [0,2pi) by definition in LorentzVectorBase
+function phi02pi(lv::LorentzVector)
+#    return mod2pi(LorentzVectorBase.phi(lv))
+    return phi(lv) < 0.0 ? phi(lv) + 2π : phi(lv)
+end
+
+function phi_mpi_pi(x)
+    twopi = 2pi
```

Comment:
yeah if we really want to minimize type...
```suggestion
    twopi = 2*oftype(x, π)
```